zerver/tests/test_auth_backends.py

```diff
@@ -216,6 +216,10 @@ def test_email_auth_backend_disabled_password_auth(self) -> None:
                                                               password,
                                                               realm=get_realm("zulip")))
 
+    def test_login_preview(self) -> None:
+        result = self.client_get('/login/?preview=True')
+        self.assertEqual(result.status_code, 200)
```

Comment:
I don't think this test has the right setup; I'm pretty sure we'd get status 200 regardless.  I think it should do this:
(0) Login
(1) Test that the login page with preview=true doesn't redirect while logged
(2) Test that the login page redirects if you're logged in

Let's assert something about the content of the login page.

There's a general principle here that @Hypro999 pointed out, but I'll reiterate it: When writing a unit test to verify your changes, you should always make sure the test fails without your code changes.  Otherwise, it's very likely you made a mistake writing the test (which happens for everyone like 50% of the time on the first try), and your test is useless or really counterproductive (since it makes one think the thing is tested, when it isn't!).